wrappers/openvino/dnn/rs-dnn-vino.cpp

```diff
@@ -0,0 +1,271 @@
+    rs2::pipeline pipe;
+    pipe.start();
+    rs2::align align_to( RS2_STREAM_COLOR );
+
+    // Start the inference engine, needed to accomplish anything. We also add a CPU extension, allowing
+    // us to run the inference on the CPU. A GPU solution may be possible but, at least without a GPU,
+    // a CPU-bound process is faster. To change to GPU, use "GPU" instead (and disable the exception):
+    openvino::Core engine;
+    openvino_helpers::error_listener error_listener;
+    engine.SetLogCallback( error_listener );
+    std::string const device_name { "CPU" };
+    engine.AddExtension( std::make_shared< openvino::Extensions::Cpu::CpuExtensions >(), device_name );
+
+    std::vector< detector_and_labels > detectors;
+    load_detectors_into( detectors, engine, device_name );
+    if( detectors.empty() )
+    {
+        LOG(ERROR) << "No detectors available in: " << fs::getcwd();
+        return EXIT_FAILURE;
+    }
+    // Look for the mobilenet-ssd so it always starts the same... otherwise default to the first detector we found
+    size_t current_detector = 0;
+    for( size_t i = 1; i < detectors.size(); ++i )
+    {
+        if( detectors[i]->pathToModel == "mobilenet-ssd.xml" )
+        {
+            current_detector = i;
+            break;
+        }
+    }
+    auto p_detector = detectors[current_detector].detector;
+    LOG(INFO) << "Current detector set to (" << current_detector+1 << ") \"" << openvino_helpers::remove_ext( p_detector->pathToModel ) << "\"";
+    auto p_labels = &detectors[current_detector].labels;
+
+    const auto window_name = "OpenVINO DNN sample";
+    cv::namedWindow( window_name, cv::WINDOW_AUTOSIZE );
+
+    cv::Mat prev_image;
+    openvino_helpers::detected_faces faces;
+    size_t id = 0;
+    uint64 last_frame_number = 0;
+    high_resolution_clock::time_point switch_time = high_resolution_clock::now();
+
+    while( cv::getWindowProperty( window_name, cv::WND_PROP_AUTOSIZE ) >= 0 )
+    {
+        // Wait for the next set of frames
+        auto frames = pipe.wait_for_frames();
+        // Make sure the frames are spatially aligned
+        frames = align_to.process( frames );
+
+        auto color_frame = frames.get_color_frame();
+        auto depth_frame = frames.get_depth_frame();
+
+        // If we only received a new depth frame, but the color did not update, continue
+        if( color_frame.get_frame_number() == last_frame_number )
+            continue;
+        last_frame_number = color_frame.get_frame_number();
+
+        auto image = frame_to_mat( color_frame );
+
+        // We process the previous frame so if this is our first then queue it and continue
+        if( ! p_detector->_request )
+        {
+            p_detector->enqueue( image );
+            p_detector->submit_request();
+            prev_image = image;
+            continue;
+        }
+
+        // Wait for the results of the previous frame we enqueued: we're going to process these
+        p_detector->wait();
```

Comment:
This seem to be a blocking code - is it possible to offload and wait for detector's result in a separate thread to get more performance ?

Reply:
Sure, but I thought that's beyond the purpose of the sample. I actually did this in the viewer.
And some steps are made to actually improve performance: the frame is sent for inference and we only wait for the result once the next frame arrives.